### DISCUSSION TITLE
Implement the health_check

### DISCUSSION THREAD

**Discussion by luoenzhen** [RDFLib/prez#308] [Q&A] [answered]
As we want to preform health checks to the triple store and/or other endpoints periodically, would it be possible to implement the health_check function please? Currently, it only returns a status "ok".

Ideally it should be configurable with an array of endpoints and a health check interval for each of the endpoints. This way, the UI can call that api and retrieve the results from the saved value stored in the server-side memory, which won't scale up with concurrent use. (e.g. 100 requests but only get one triggered).

Thank you in advance.

**Comment by recalcitrantsupplant** [comment, marked as answer]
I'm not sure it's sensible to have prez orchestrate healthchecks, it does the startup one more as a convenience for the app to exit quickly in the case there's some connection issue. You can implement healthchecks outside of prez, if you're using a cloud provider they offer this functionality in most products. Not sure what the relation from healthchecks to scaling is.
